### ISSUE TITLE
Axis 75 kHz data quality

### ISSUE THREAD

**Issue by kurtisanstey** [kurtisanstey/project#15]
Old & new data are the same pre-May 2013. Half-day March 2013:
![image](https://user-images.githubusercontent.com/57200139/106121897-edad6e00-610c-11eb-905f-a1dfad96e05c.png)
___________________________________________________________________________________________
Old & new data are the same pre-May 2013. Week in March 2013:
![image](https://user-images.githubusercontent.com/57200139/106121925-fa31c680-610c-11eb-97ff-be332689bfa6.png)
___________________________________________________________________________________________
Old & new data are the same pre-May 2013. Everything before fix (July 2012- May 2013):
![image](https://user-images.githubusercontent.com/57200139/106121965-061d8880-610d-11eb-9803-9c7f4b062960.png)
___________________________________________________________________________________________
New data, half-day March 2014 (after fix date):
![image](https://user-images.githubusercontent.com/57200139/106183887-aeefd600-6155-11eb-88f7-4f62d09fa80f.png)
Old data, half-day March 2014:
![image](https://user-images.githubusercontent.com/57200139/106183348-f6299700-6154-11eb-8306-002104a16944.png)
___________________________________________________________________________________________
New data, week in March 2014 (after fix date):
![image](https://user-images.githubusercontent.com/57200139/106184467-6c7ac900-6156-11eb-96b1-c2f8990782f7.png)
Old data, week in March 2014:
![image](https://user-images.githubusercontent.com/57200139/106184511-7bfa1200-6156-11eb-9f04-1dba228ac75e.png)

**Comment by jklymak** [comment]
I can't tell what is what in the above plots - which one is supposed to have been improved?  what is the before and after?

**Comment by kurtisanstey** [comment]
Sorry @jklymak, this post wasn't ready for you, yet. Just something I was working on when I couldn't sleep. Do you get notifications even if I don't tag you?

Removing the beam didn't seem to help much, it just lessened the velocity amplitudes a bit, and they only removed it from May 2013 onward. I still see strong vertical banding, and weaker horizontal banding, in the winter/spring of both 2013 (before fix) and 2014 (after fix). I may need them to check the data again, regarding these issues, unless there's some way to filter it myself with backscatter/correlation info when they provide that in the NetCDF files. 

It's a really interesting feature near the rim of the canyon, in the winter/spring, if I can get this data looking good.

**Comment by kurtisanstey** [comment]
- [x] Check to see if the new data was processed the same as the old data (rotation, etc.) to account for velocity amplitude differences.
- [x] Double-check if bands span through depth. 
- [x] Adjust colour bars to see if it's possible the blobs are real. 
- [x] Email ONC regarding the vertical bands, if they're 'really' present.

**Comment by kurtisanstey** [comment]
See email chain for details.

Ultimately, correlation and echo intensity plots show poor data above about -650 m depth. Will use data below that, for now, and write an algorithm to determine optimal depth based on thresholds, soon. ONC notified of results.

Ze confirmed that my plots were correct, and that there are some issues with correlation saturation 'jumping around' to different beams. They are going to do a complete check of the dataset so they can adjust beams accordingly, and add the metadata.

**Comment by kurtisanstey** [comment]
Depth quality algorithm has a rough form, and will revisit if there's time, later. For now, a conservative estimate of -700 m will be used as a threshold depth, estimated from the correlation strength and echo intensity plots. 

ONC has updated the NetCDF metadata, and turned some beams on/off for problem periods. Ze has provided a few comparison plots, and noted that these periods can be checked in each instruments Additional Information fields. They will provide a thorough comparison when they have some more time.

Data is currently processing, and will hopefully be ready to check for our meeting.

**Comment by kurtisanstey** [comment]
Check for added metadata, and run through new analysis process for updated plots. Then interpret them!

**Comment by kurtisanstey** [comment]
Some samples from the updated Axis 75 kHz data, after the ONC changes.

Metadata has been added:
![image](https://user-images.githubusercontent.com/57200139/108259947-8df82080-7116-11eb-9ede-423ad685d73a.png)

And a few before and after plots:

**Comment by kurtisanstey** [comment]
Some samples from the updated Axis 75 kHz data, after the ONC changes. A more comprehensive update is in the Analysis document, and will be reflected in the committee meeting presentation.

Metadata has been added:
![image](https://user-images.githubusercontent.com/57200139/108259947-8df82080-7116-11eb-9ede-423ad685d73a.png)

Before:
![image](https://user-images.githubusercontent.com/57200139/108409882-96665f00-71db-11eb-9617-17cf4a83b7be.png)
After:
![image](https://user-images.githubusercontent.com/57200139/108409945-a9792f00-71db-11eb-85e5-cb0018a5b6e9.png)

Before:
![image](https://user-images.githubusercontent.com/57200139/108409652-58693b00-71db-11eb-988f-068a642004c3.png)
After:
![image](https://user-images.githubusercontent.com/57200139/108409730-6f0f9200-71db-11eb-9226-7681a4f9ec25.png)

Before:
![image](https://user-images.githubusercontent.com/57200139/108409280-e4c72e00-71da-11eb-9cf1-55e2a8cdb703.png)
After:
![image](https://user-images.githubusercontent.com/57200139/108409379-032d2980-71db-11eb-8fb6-3c74088a779c.png)

**Comment by kurtisanstey** [comment]
@jklymak 

Some samples from the updated Axis 75 kHz data, after the ONC changes. These were all run through the same processes, so any differences are in the data. I have not done a raw data comparison for these datasets. A more comprehensive update is in the Analysis document, and will be reflected in the committee meeting presentation.
________________________

Metadata has been added:

![image](https://user-images.githubusercontent.com/57200139/108259947-8df82080-7116-11eb-9ede-423ad685d73a.png)
________________________

There is some difference in the velocity magnitudes, mostly in the along-slope direction, though the features are similar. This is more pronounced than in the initial 4-beam to 3-beam adjustment.

Before:
![image](https://user-images.githubusercontent.com/57200139/108411964-142b6a00-71de-11eb-8ab1-15c5cab3621b.png)
After:
![image](https://user-images.githubusercontent.com/57200139/108409945-a9792f00-71db-11eb-85e5-cb0018a5b6e9.png)
________________________

There is more pronounced high frequency activity in the along-slope spectrograms.

Before:
![image](https://user-images.githubusercontent.com/57200139/108411090-104b1800-71dd-11eb-8e44-2d1297d9d256.png)
After:
![image](https://user-images.githubusercontent.com/57200139/108409730-6f0f9200-71db-11eb-9226-7681a4f9ec25.png)
________________________

The heightened magnitudes in the along-slope direction are reflected in the spectra.

Before:
![image](https://user-images.githubusercontent.com/57200139/108410828-be09f700-71dc-11eb-9a25-d49ea269307d.png)
After:
![image](https://user-images.githubusercontent.com/57200139/108409379-032d2980-71db-11eb-8fb6-3c74088a779c.png)

**Comment by kurtisanstey** [comment]
Create a Python notebook for Jody re:
- [x] Need actual data for correlation, intensity, velocity error, etc., not just metadata.
- [x] Comparison of raw data in 2014 between datasets.
- [x] Previous comparisons between datasets.
- [x] Correlation spot checks.
- [x] Additional Attributes on ONC website showing ExcludeADCPbeam field.
- [x] Timeline of instruments with dates and processing notes.

**Comment by kurtisanstey** [comment]
Jody if you see this I'm just compiling the information here, and adding it to a Python notebook for you, now. I'll send that to you when it's done, soon.
__________________

**Current issues**
- New NetCDF data have updated correlation and error velocity threshold values in the metadata, but still do not contain the correlation, intensity, or error velocity data, for plotting. These should be added to the NetCDF files.
- Both averaged and raw comparisons show an increase in velocity magnitudes / variance in the new data, so whatever changes were made do make a difference. We'd like a record of any changes made, and why they were made.

**Fixed issues**
- Correlation and intensity spot checks still show correlation saturation issues, but the ExcludeADCPbeamNumber fields on the ONC website say those beams were turned off for those instruments, so this issue seems to be accounted for. 
___________________

**Data availability and instrument timeline**

![image](https://user-images.githubusercontent.com/57200139/108773726-a3ed5300-7513-11eb-9565-d464e2f06d4f.png)

**Axis** |   |   |   |   |  
-- | -- | -- | -- | -- | --
Device ID | Deployed | Removed | ADCP type | Important notes | ExcludeADCPbeamNumber
23064 | 2012-06-02 | 2014-05-05 | RDI 75 kHz Workhorse Long Ranger | Had later issues with beam 1   associated with faulty DSP board (2019-05-01), long range interference issues   (2013-05-23), and 2-hour interference issues (2013-01-16). | Beam 2, from 2013-09-01
23065 | 2014-05-08 | 2015-09-17 | RDI 75 kHz Workhorse Long Ranger | Interferences noted in data   (2014-07-10), removed for maintenance in 2015. Had later issues with beam 1   associated with faulty DSP board (2019-05-01) - why is this listed for both   of these RDI instruments? | Beam 1, from 2014-05-08
  | 2016-06-14 | 2018-06-22 |   |   |  
22925 | 2016-06-14 | 2018-06-22 | Nortek 55 kHz Signature55 | Had later issues with beam 3   having excessive noise (2019-01-22). | n/a
24420 | 2018-06-22 | n/a | Nortek 55 kHz Signature55 | Lost connection (2020-06-01). | n/a
  |   |   |   |   |  
**Upper Slope** |   |   |   |   |  
Device ID | Deployed | Removed | ADCP type | Important notes | ExcludeADCPbeamNumber
12108 | 2009-07-26 | 2019-05-16 | RDI 75 kHz Workhorse Long Ranger | Nothing of note. | None
23065 | 2019-08-26 | n/a | RDI 75 kHz Workhorse Long Ranger | Lost connection (2020-06-01). | Beam 1, from 2014-05-08
___________________

**New 3-beam NetCDF metadata updates**

New NetCDF data have updated correlation and error velocity threshold values in the metadata, but do not contain the correlation, intensity, or error velocity data, for plotting.

The only variables available are *u*, *v*, *w*, and temperature:

![image](https://user-images.githubusercontent.com/57200139/108610829-9b6b1000-738d-11eb-814e-6953f3732bff.png)

Additions to metadata:

![image](https://user-images.githubusercontent.com/57200139/108610865-fb61b680-738d-11eb-8c30-57253e5c72d2.png)
___________________

**Single-day velocity comparison (*u* velocities; July 1, 2014), 15-minute averaged data** 

- 'New 3-beam data' was downloaded on February 17, 2021, and includes updated metadata; 'Previous 3-beam data' was downloaded on January 27, 2021, and included beam-2 being shut off for all data beyond September 1, 2013. 
- The new averaged data shows increased velocity magnitudes, reflected in greater variance.

New 3-beam data (NetCDF), variance @ -800 m = 0.0005048 (m/s)^2:
![image](https://user-images.githubusercontent.com/57200139/108579782-60110880-72dd-11eb-9cdf-e3833bf9dddf.png)
Previous 3-beam data (NetCDF), variance @ -800 m = 0.0004001 (m/s)^2:
![image](https://user-images.githubusercontent.com/57200139/108579708-08729d00-72dd-11eb-8c10-c6f8cb5c2a64.png)
___________________

**Single-day velocity comparison (*u* velocities; July 1, 2014), raw 2-second data** 

The new raw data shows the same increase in velocity magnitudes, and is more visually apparent in these plots than for the averaged data - the new data is much darker than the previous data. I confirmed the magnitude increase in the new raw data in both NetCDF and MAT format. It is again reflected in increased variance.

New 3-beam data (NetCDF; also confirmed in MAT), variance @ -800 m = 0.0097734 (m/s)^2:
![image](https://user-images.githubusercontent.com/57200139/108610618-eedc5e80-738b-11eb-9a97-330505b4fd0c.png)
Previous 3-beam data (MAT), variance @ -800 m = 0.0046474 (m/s)^2:
![image](https://user-images.githubusercontent.com/57200139/108610294-77a5cb00-7389-11eb-9554-3cda128c5b89.png)
___________________

**Correlation and intensity spot check - March 1, 2014**

There is no noticeable difference between the old and new data, and the issue with Beam 2 correlation saturation is still present. ONC notes this is accounted for in ExcludeADCPbeamNumber information, online.

New 3-beam data. Beam 2 still shows the correlation saturation issue from the old data, and intensity still appears fine:

![image](https://user-images.githubusercontent.com/57200139/108611313-60b7a680-7392-11eb-834c-db29e10035c2.png)
![image](https://user-images.githubusercontent.com/57200139/108611321-72994980-7392-11eb-86a3-dee7e74ece01.png)

Previous 3-beam data. Beam 2 shows correlation saturation, and intensity seems fine:

![image](https://user-images.githubusercontent.com/57200139/108581001-f98ee900-72e2-11eb-92cc-2a252764d60c.png)
![image](https://user-images.githubusercontent.com/57200139/108581008-03b0e780-72e3-11eb-9b02-ae9a072fb6af.png)
___________________

**Correlation and intensity spot check - July 1, 2014**

There is no noticeable difference between the old and new data, and the issue with Beam 1 correlation saturation is still present. ONC notes this is accounted for in ExcludeADCPbeamNumber information, online.

New 3-beam data. Beam 1 still shows the correlation saturation issue from the old data, and intensity still appears fine:

![image](https://user-images.githubusercontent.com/57200139/108611201-141f9b80-7391-11eb-97f3-acd0a294a6ad.png)
![image](https://user-images.githubusercontent.com/57200139/108611215-39aca500-7391-11eb-9da9-29d593b0b221.png)

Previous 3-beam data. Beam 1 shows correlation saturation, and intensity seems fine:

![image](https://user-images.githubusercontent.com/57200139/108580662-525d8200-72e1-11eb-96e0-6ca5e0307b6d.png)
![image](https://user-images.githubusercontent.com/57200139/108580700-7d47d600-72e1-11eb-8fa9-536439bffb25.png)

**Comment by kurtisanstey** [comment]
@jklymak sending this record to you as a Python notebook in an email. Will leave this issue open for now, for updates from ONC.

**Comment by jklymak** [comment]
Sorry Kurtis,  I'm finding the above *really* hard to parse.  What is your conclusion?

**Comment by kurtisanstey** [comment]
@jklymak , the above is just me collecting the information. I'm sending you an email with the conclusions and a copy of the Notebook file for reference.

**Comment by kurtisanstey** [comment]
@jklymak will follow up about:

- Adding the correlation and intensity data to the NetCDF files (not just the single threshold value in the metadata).
- Documenting when changes are made to instruments to show the effects on the data, and what the reasons were, as it is not easy to piece it together from what is online (and sometimes all the details aren't available online).
